ui/app/services/wizard.js

```diff
@@ -0,0 +1,262 @@
+          break;
+        case 'showTutorialWhenAuthenticated':
+          this.set('showWhenUnauthenticated', false);
+          break;
+        case 'showTutorialAlways':
+          this.set('showWhenUnauthenticated', true);
+          break;
+        case 'continueFeature':
+          this.transitionFeatureMachine(this.get('featureState'), 'CONTINUE', this.get('componentState'));
+          break;
+        default:
+          break;
+      }
+    }
+  },
+
+  handleDismissed() {
+    this.storage().removeItem(FEATURE_STATE);
+    this.storage().removeItem(FEATURE_LIST);
+    this.storage().removeItem(COMPONENT_STATE);
+  },
+
+  saveFeatures(features) {
+    this.set('featureList', features);
+    this.saveExtState(FEATURE_LIST, this.get('featureList'));
+    this.buildFeatureMachine();
+  },
+
+  buildFeatureMachine() {
+    if (this.get('featureList') === null) {
+      return;
+    }
+    this.startFeature();
+    if (this.storageHasKey(FEATURE_STATE)) {
+      this.saveState('featureState', this.getExtState(FEATURE_STATE));
+    }
+    this.saveExtState(FEATURE_STATE, this.get('featureState'));
+    let nextFeature =
+      this.get('featureList').length > 1 ? this.get('featureList').objectAt(1).capitalize() : 'Finish';
+    this.set('nextFeature', nextFeature);
+    let next;
+    if (this.get('currentMachine') === 'secrets' && this.get('featureState') === 'display') {
+      next = FeatureMachine.transition(this.get('featureState'), 'REPEAT', this.get('componentState'));
+    } else {
+      next = FeatureMachine.transition(this.get('featureState'), 'CONTINUE', this.get('componentState'));
+    }
+    this.saveState('nextStep', next.value);
+    let stateNodes = FeatureMachine.getStateNodes(this.get('featureState'));
+    this.executeActions(stateNodes.reduce((acc, node) => acc.concat(node.onEntry), []));
+  },
+
+  startFeature() {
+    const FeatureMachineConfig = MACHINES[this.get('featureList').objectAt(0)];
+    FeatureMachine = Machine(FeatureMachineConfig);
+    this.set('currentMachine', this.get('featureList').objectAt(0));
+    this.saveState('featureState', FeatureMachine.initialState);
+  },
+
+  completeFeature() {
+    let features = this.get('featureList');
+    let done = features.shift();
+    if (!this.getExtState(COMPLETED_FEATURES)) {
+      let completed = [];
+      completed.push(done);
+      this.saveExtState(COMPLETED_FEATURES, completed);
+    } else {
+      this.saveExtState(COMPLETED_FEATURES, this.getExtState(COMPLETED_FEATURES).toArray().addObject(done));
+    }
+
+    this.saveExtState(FEATURE_LIST, features.length ? features : null);
+    this.storage().removeItem(FEATURE_STATE);
+    if (features.length > 0) {
+      this.buildFeatureMachine();
+    } else {
+      this.storage().removeItem(FEATURE_LIST);
+      FeatureMachine = null;
+      this.transitionTutorialMachine(this.get('currentState'), 'DONE');
+    }
+  },
+
+  storage() {
+    return getStorage();
+  },
+});
```

Comment:
Some of these methods are rather dense, but I'm not sure if they need more comments or if I just need to know more about xstate.